docs/en/advanced-concepts/migrating-to-appium-2.0.md

```diff
@@ -127,19 +127,129 @@ Apart from the breaking changes mentioned above, in this section is a list of so
+
+### Suggested capabilities
+
+In addition to the standard `platformName`, `appium:deviceName`, `appium:automationName`, and `appium:platformVersion`, we recommend adopting the capability `$cloud:appiumOptions`, where the label `$cloud` is not meant to be interpreted literally but instead should be replaced by your vendor prefix (so for HeadSpin it would be `headspin`, Sauce Labs it would be `sauce`, and BrowserStack it would be `browserstack`, to name just a few examples). The `$cloud:appiumOptions` capability would itself be a JSON object, with the following internal keys:
+
+|Capability|Used for|Example|
+|----------|-------|-------|
+|`version`|Designating which version of the Appium server is used to host and manage drivers. If ommitted, behavior left up to the provider, but the recommendation would be to provide the latest official version.|`2.0.0`|
+|`automationVersion`|Designating which version of the specified driver should be used.|`1.55.2`|
+|`automation`|Designating a custom driver to use (see below for more info). This would override `appium:automationName` and `$cloud:automationVersion`|`{"name": "org/custom-driver", "source": "github", "package": "custom-driver"}`|
+|`plugins`|Designating the list of plugins (and potentially versions of plugins) to be activated (see below for more info).|`["images", "universal-xml"]`|
+
+### Basic example
+
+Appium extensions (drivers and plugins) have a set of properties that specify where they can be installed from. Cloud providers are obviously under no obligation to provide support for arbitrarily specified extensions, seeing as these may represent untrusted code running in a managed environment. In the case where arbitrary extensions are not supported, the `appium:automationName`, `$cloud:automationVersion`, and `$cloud:appiumPlugins` capabilities should be sufficient. See the following JSON object representing capabilities for a session:
+
+```json
+{
+    "platformName": "iOS",
+    "appium:platformVersion": "14.4",
+    "appium:deviceName": "iPhone 11",
+    "appium:app": "Some-App.app.zip",
+    "appium:automationName": "XCUITest",
+    "$cloud:appiumOptions": {
+        "appiumVersion": "2.0.0",
+        "automationVersion": "3.52.0",
+        "plugins": ["images"],
+    }
+}
+```
```

Comment:
> `$cloud:automationVersion`, and `$cloud:appiumPlugins`

Am I assuming right that Appium understands and treats the following cap examples in the same way:

```js
{
    "platformName": "iOS",
    "appium:platformVersion": "14.4",
    "appium:deviceName": "iPhone 11",
    "appium:app": "Some-App.app.zip",
    "appium:automationName": "XCUITest",
    "$cloud:appiumVersion": "2.0.0",
    "$cloud:automationVersion": "3.52.0",
    "$cloud:plugins": ["images"]
}
```
vs
```js
{
    "platformName": "iOS",
     "appium:options": {
        "platformVersion": "14.4",
        "deviceName": "iPhone 11",
        "app": "Some-App.app.zip",
        "automationName": "XCUITest"
    },
    "$cloud:appiumOptions": {
        "appiumVersion": "2.0.0",
        "automationVersion": "3.52.0",
        "plugins": ["images"],
    }
}
```
If we talk about `appium:options` a bit further above it seems confusing to use the other style in this example. If both ways are supported (thinking about backwards compat to Appium 1) I suggest to specially mention the differences.

Reply:
Hmm, that is what I tried to do with:

> The previous example still looks a bit disorganized, so of course we also recommend that cloud providers support the `appium:options` capability as detailed above,

Can you suggest better phrasing?

Reply:
Re-reading it now clears up the confusion. I think I missed the fact that `appium:automationName` is currently the way people pass in this information.